From 8575592802b9527fe0f7cf19e9cb7159c9aa5121 Mon Sep 17 00:00:00 2001
From: Sumedh Sen <sumedhsen@google.com>
Date: Thu, 16 Jan 2025 11:38:56 -0800
Subject: [PATCH] Get calling package name using a reliable API

getCallingPackage() is unreliable due to reasons listed here:
go/getcallingpackage-misuse. Replace it with getLaunchedFromPackage

Test: Manual. Install an APK and ensure app is installed
Bug: 364604008
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:1ab193cb57c22605e6cffb52ba1b222d8a2637c5)
Merged-In: If94ea6566fa3f3b461efaeffff7b074693003c9a
Change-Id: If94ea6566fa3f3b461efaeffff7b074693003c9a
---
 .../src/com/android/packageinstaller/InstallStart.java          | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/packages/PackageInstaller/src/com/android/packageinstaller/InstallStart.java b/packages/PackageInstaller/src/com/android/packageinstaller/InstallStart.java
index 229b7a73a2a3..e70acc314658 100644
--- a/packages/PackageInstaller/src/com/android/packageinstaller/InstallStart.java
+++ b/packages/PackageInstaller/src/com/android/packageinstaller/InstallStart.java
@@ -68,7 +68,7 @@ public class InstallStart extends Activity {
         mUserManager = getSystemService(UserManager.class);
 
         Intent intent = getIntent();
-        String callingPackage = getCallingPackage();
+        String callingPackage = getLaunchedFromPackage();
         String callingAttributionTag = null;
 
         final boolean isSessionInstall =
-- 
2.49.0.395.g12beb8f557-goog

